engine/schema/src/main/java/com/cloud/upgrade/SystemVmTemplateRegistration.java

```diff
@@ -595,6 +599,20 @@ public void updateSystemVMEntries(Long templateId, Hypervisor.HypervisorType hyp
         vmInstanceDao.updateSystemVmTemplateId(templateId, hypervisorType);
     }
 
+    public void updateSystemVmTemplateGuestOsId() {
+        String systemVmGuestOsName = "Debian GNU/Linux 12 (64-bit)";
+        GuestOSVO guestOS = guestOSDao.findOneByDisplayName(systemVmGuestOsName);
+        if (guestOS != null) {
+            LOGGER.debug("Updating SystemVM Template Guest OS [{}] id", systemVmGuestOsName);
+            SystemVmTemplateRegistration.LINUX_12_ID = Math.toIntExact(guestOS.getId());
+            hypervisorGuestOsMap.put(Hypervisor.HypervisorType.KVM, LINUX_12_ID);
+            hypervisorGuestOsMap.put(Hypervisor.HypervisorType.XenServer, LINUX_12_ID);
+            hypervisorGuestOsMap.put(Hypervisor.HypervisorType.VMware, LINUX_12_ID);
+            hypervisorGuestOsMap.put(Hypervisor.HypervisorType.Hyperv, LINUX_12_ID);
+            hypervisorGuestOsMap.put(Hypervisor.HypervisorType.LXC, LINUX_12_ID);
+        }
+    }
```

Comment:
is there any way we can detect? this remains a maintenance issue after this change.